### PR TITLE
doc: link auth design in doc

### PR DESCRIPTION
**Pull request by xiang90**
/cc @mitake @heyitsanthony

### CODE REVIEW

File: Documentation/docs.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ To learn more about the concepts and internals behind etcd, read the following p
  - [Why etcd][why] (TODO)
  - [Understand data model][data_model]
  - [Understand APIs][understand_apis]
+ - [Understand the design of the Auth subsystem][understand_auth]
```
Comment (heyitsanthony) [inline Documentation/docs.md]: this is more of an internals thing?

Comment (xiang90) [inline Documentation/docs.md]: The learning section will eventually contain more and more internal design stuff. I would encourage people who want to learn more about etcd to read these things.

Comment (xiang90) [inline Documentation/docs.md]: but yea, do you have any suggestions to group this better? my main goal is to make the internal design doc more discoverable. so people who want to learn more can find them.

Comment (heyitsanthony) [inline Documentation/docs.md]: something like this?

* Why etcd?
* Data model
* APIs
* Internals
  * Raft
  * Write-ahead log
  * Key-value backend
  * Auth subsystem

Comment (xiang90) [inline Documentation/docs.md]: sgtm. i will re-org this.

**Comment by xiang90** [comment]
@heyitsanthony fixed. PTAL.

**Comment by fanminshi** [comment]
lgtm

**Comment by mitake** [comment]
@xiang90 sorry for my late reply. The change seems good to me.

**Pull request merged**